data/src/test/java/com/linkedin/data/schema/TestSchemaToPdlEncoder.java

```diff
@@ -22,19 +22,38 @@
 import java.io.StringWriter;
 import java.util.HashMap;
 import java.util.Map;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
 import static org.junit.Assert.*;
 
 
 public class TestSchemaToPdlEncoder
 {
-  @Test
-  public void testEncodeRecordWithEmptyDataMapInProperty() throws IOException
+
+  @DataProvider
+  private static Object[][] recordSchemaProvider()
+  {
+    DataMap properties1 = new DataMap();
+    properties1.put("empty", new DataMap());
+    DataMap properties2 = new DataMap();
+    properties2.put("validate", properties1);
+    DataMap properties3 = new DataMap();
+    DataMap nestedMap = new DataMap(properties2);
+    nestedMap.putAll(properties1);
+    properties3.put("nested", nestedMap);
+    return new Object[][]{{properties1, "namespace com.linkedin.test\n" + "\n" + "@empty = {}\nrecord RecordDataSchema {}"},
+        {properties2, "namespace com.linkedin.test\n" + "\n" + "@validate.empty = {}\n" + "@empty = {}\nrecord RecordDataSchema {}"},
+        {properties3, "namespace com.linkedin.test\n" + "\n" + "@nested = {\"validate\":{\"empty\":{}},\"empty\":{}}\n"
+            + "@empty = {}\n" + "record RecordDataSchema {}"}};
+  }
+
+  @Test(dataProvider = "recordSchemaProvider")
+  public void testEncodeRecordWithEmptyDataMapInProperty(Map<String, Object> properties, String pdlString) throws IOException
```

Comment:
By adding more test cases that don't apply to the original intent of the test method, this becomes more confusing. I'd suggest making this more generalized of a method which just tests custom property encoding.

However, this brings up another interesting point. This test assumes the ordering of properties, which is currently NOT deterministic. I'd suggest keeping the tests right now limited to only one property per test case, and adding a TODO to be addressed by whoever implements deterministic ordering. _In addition_, this sort of testing should be done by `PdlEncoderTest`, which makes testing the encoder much easier. You can avoid defining the schema as a string literal.

Reply:
I thought of writing it in PdlEncoderTest but then its in a different module, and also uses SchemaToPdlEncoder.
Now, I have added a new test class for TestPdlEncoder in same package.

Regarding the deterministic ordering I have removed the last test for now.